### ISSUE TITLE
k-fold validation for bootstrapping

### ISSUE THREAD

**Issue by TMondain** [robboyd/soaR#5]
Using k-fold validation for bootstrapping might be a problem if you want to have >5 bootstraps. Because as you increase the value of k (to increase bootstrap number) you are also reducing the size of the testing dataset relative to the training dataset. I don't know if/how this will affect the evaluation of models, I haven't done any tests yet to check.

Have been thinking about this a bit more and it shouldn't matter if you only want to have 5 bootstraps of each model because the testing data will only be 1/5th of the data as a whole, but if people want to run 20/30/40 bootstraps then it would probably affect the evaluation process of the function quite a lot due to the potential of overfitting.

**Comment by robboyd** [comment]
Good point @TMondain. Should change this so that we resample with replacement maybe

**Comment by TMondain** [comment]
Yeah, I think the easiest would be to just have a bootstrapping process separate to the evaluation part of the function? Then we don't have to worry about how the bootstrapping affects the model evaluation.

**Comment by robboyd** [comment]
Think I agree, but would like to retain the option to kill two birds with one stone and get uncertainty from the cross validation. Need to be as efficient as possible for my high throughput stuff

**Comment by TMondain** [comment]
That makes sense! We could have the two methods coded in the function and just say if the number of bootstraps > 5 (or some other number) then the bootstrapping is done separate from the evaluation process.  Just need to explain it clearly in the documentation I suppose!
